xarray/core/indexes.py

```diff
@@ -632,7 +632,7 @@ def from_variables(
         # the checks below.
 
         # preserve wrapped pd.Index (if any)
-        data = getattr(var._data, "array", var.data)
+        data = getattr(var._data, "array") if hasattr(var._data, "array") else var.data
```

Comment:
can you explain what is going on here? Why does `var.data` trigger an additional load? Where does our cache mechanism come in here?

Reply:
`var._data` contains `array(array(array(...)))` . We load these on the first access to `.data` and `.values`

This is a regression from Anderson's work on indexing I think. Something in the pipeline was reading this earlier and so it was fine at this point. Now we must be more careful about loading and this line was loading it again.

Reply:
Am I being slow or do these two lines have exactly the same semantics?

Reply:
Yes but else conditions don't get evaluated unless necessary

Reply:
Ah sorry, I was being slow

Reply:
Sorry the ternary thing is a bit cute :) The comment makes it better.

Reply:
`hasattr(obj, name)` is currently an alias of
```python
try:
    getattr(obj, name)
    return True
except AttributeError:
    return False
```
so it still will be evaluated. If you don't want that, use inspect.